crates/driver/src/infra/api/routes/solve/dto/auction.rs

```diff
@@ -64,24 +64,34 @@ impl Auction {
                         } else {
                             competition::order::Partial::No
                         },
-                        pre_interactions: order
-                            .pre_interactions
-                            .into_iter()
-                            .map(|interaction| eth::Interaction {
-                                target: interaction.target.into(),
-                                value: interaction.value.into(),
-                                call_data: interaction.call_data.into(),
-                            })
-                            .collect(),
-                        post_interactions: order
-                            .post_interactions
-                            .into_iter()
-                            .map(|interaction| eth::Interaction {
-                                target: interaction.target.into(),
-                                value: interaction.value.into(),
-                                call_data: interaction.call_data.into(),
-                            })
-                            .collect(),
+                        pre_interactions: match order.executed.is_zero() {
```

Comment:
Not sure if you think this is nicer or not, but:

```suggestion
                        pre_interactions: order.executed.is_zero()
                            .then(|| order.pre_interactions...)
                            .unwrap_or_default(),
```

Reply:
I moved that logic into the `autopilot` and used your suggestion there. 👍